### PR TITLE
SECURITY.md updated to Eclipse's policy, rust toolchain update

### LINKED ISSUE #281: [PBI] Update SECURITY file to point to Eclipse Foundation's Security Policy and contact
## Description

SECURITY.md points to a Microsoft Security policy, it should be updated to include the Eclipse Foundation's Security policy and contact instead

## Acceptance criteria

- [ ] SECURITY.md updated with Eclipse Foundation information

### PR DESCRIPTION
**Pull request by ladatz**
Resolves #281 

## Motivation and Context

For legacy reasons the repository had Microsoft's security policy listed. We need to update it to reflect Eclipse Foundation's policy.

## Description

- Use Eclipse foundations' security policy in SECURITY.md 
- Update rust toolchain version to 1.72.1

### CODE REVIEW

**Review by devkelley — APPROVED** [review]

File: SECURITY.md [hunk 1]
```diff
@@ -1,41 +1,10 @@
...
 
-Microsoft takes the security of our software products and services seriously, which includes all source code repositories managed through our GitHub organizations, which include [Microsoft](https://github.com/Microsoft), [Azure](https://github.com/Azure), [DotNet](https://github.com/dotnet), [AspNet](https://github.com/aspnet), [Xamarin](https://github.com/xamarin), and [our GitHub organizations](https://opensource.microsoft.com/).
+* <https://www.eclipse.org/security>
 
-If you believe you have found a security vulnerability in any Microsoft-owned repository that meets [Microsoft's definition of a security vulnerability](https://docs.microsoft.com/en-us/previous-versions/tn-archive/cc751383(v=technet.10)), please report it to us as described below.
+## Reporting a Vulnerability
 
-## Reporting Security Issues
-
-**Please do not report security vulnerabilities through public GitHub issues.**
```
Comment (ashbeitz) [inline SECURITY.md:11]: I think that it would be good to keep this.

File: SECURITY.md [hunk 2]
```diff
@@ -1,41 +1,10 @@
...
-
-Instead, please report them to the Microsoft Security Response Center (MSRC) at [https://msrc.microsoft.com/create-report](https://msrc.microsoft.com/create-report).
-
-If you prefer to submit without logging in, send email to [secure@microsoft.com](mailto:secure@microsoft.com).  If possible, encrypt your message with our PGP key; please download it from the [Microsoft Security Response Center PGP Key page](https://www.microsoft.com/en-us/msrc/pgp-key-msrc).
-
-You should receive a response within 24 hours. If for some reason you do not, please follow up via email to ensure we received your original message. Additional information can be found at [microsoft.com/msrc](https://www.microsoft.com/msrc).
-
-Please include the requested information listed below (as much as you can provide) to help us better understand the nature and scope of the possible issue:
```
Comment (ashbeitz) [inline SECURITY.md:19]: I think that it would be good to keep this list.

File: SECURITY.md [hunk 3]
```diff
@@ -1,41 +1,10 @@
...
-
-* Type of issue (e.g. buffer overflow, SQL injection, cross-site scripting, etc.)
-* Full paths of source file(s) related to the manifestation of the issue
-* The location of the affected source code (tag/branch/commit or direct URL)
-* Any special configuration required to reproduce the issue
-* Step-by-step instructions to reproduce the issue
-* Proof-of-concept or exploit code (if possible)
-* Impact of the issue, including how an attacker might exploit the issue
-
-This information will help us triage your report more quickly.
```
Comment (ashbeitz) [inline SECURITY.md:29]: I think that it would be good to keep this.

File: SECURITY.md [hunk 4]
```diff
@@ -1,41 +1,10 @@
...
-
-If you are reporting for a bug bounty, more complete reports can contribute to a higher bounty award. Please visit our [Microsoft Bug Bounty Program](https://microsoft.com/msrc/bounty) page for more details about our active programs.
-
-## Preferred Languages
```
Comment (ashbeitz) [inline SECURITY.md:33]: I think that it would be good to keep this section.

**Review by ashbeitz — COMMENTED** [review]
A few suggestions.  Please take a look.

**Comment by fprezado** [comment]
thanks for the comments and review @ashbeitz. wWe will follow the template that Eclipse Foundation recommends in their handbook and make the update based on it. This is a useful reference for how to customize the template - https://blogs.eclipse.org/post/marta-rybczynska/securitymd-should-i-have-it

File: SECURITY.md (hunk 4 above)
Comment (ladatz) [reply SECURITY.md:33]: closing based on Filipe's comment

File: SECURITY.md (hunk 3 above)
Comment (ladatz) [reply SECURITY.md:29]: closing based on Filipe's comment

File: SECURITY.md (hunk 2 above)
Comment (ladatz) [reply SECURITY.md:19]: closing based on Filipe's comment

File: SECURITY.md (hunk 1 above)
Comment (ladatz) [reply SECURITY.md:11]: closing based on Filipe's comment

**Pull request merged**